code/batchactions/CMSBatchActions.php

```diff
@@ -68,6 +68,51 @@ public function applicablePages($ids) {
 }
 
 /**
+ * Batch restore of pages
+ */
+class CMSBatchAction_Restore extends CMSBatchAction {
+	
+	public function getActionTitle() {
+		return _t('CMSBatchActions.RESTORE', 'Restore');
+	}
+
+	public function run(SS_List $pages) {
+		// Sort pages by depth
+		$pageArray = $pages->toArray();
+		// because of https://bugs.php.net/bug.php?id=50688
+		foreach($pageArray as $page) {
+			$page->getPageLevel();
```

Comment:
This solution attempts to pre-load the page level calculation so that usort doesn't hit the `PHP Warning:  usort(): Array was modified by the user comparison function` error